### PR TITLE
Use tmpdir for more files generated during tests

### PR DESCRIPTION
**Pull request by tgaddair**
We should try and fine a way to do this at a global level for all tests, probably by mocking `get_output_directory` in some form.

### CODE REVIEW

File: tests/ludwig/augmentation/test_augmentation_pipeline.py [hunk 1]
```diff
@@ -139,14 +139,14 @@ def run_augmentation_training(
...
-    return model
-
+    with tempfile.TemporaryDirectory() as tmpdir:
+        model = LudwigModel(config, logging_level=logging.INFO)
+        model.experiment(
+            dataset=train_fp,
+            skip_save_processed_input=True,
+            skip_save_model=True,
+            output_directory=os.path.join(tmpdir, "output"),
+        )
```
Comment (arnavgarg1) [inline tests/ludwig/augmentation/test_augmentation_pipeline.py:149]: tmpdir fixture?

Comment (justinxzhao) [reply tests/ludwig/augmentation/test_augmentation_pipeline.py:149]: Can the pytest fixture be used in a non-"test_" function?

**Review by justinxzhao — APPROVED** [review]

File: tests/ludwig/augmentation/test_augmentation_pipeline.py (hunk 1 above)
Comment (tgaddair) [reply tests/ludwig/augmentation/test_augmentation_pipeline.py:149]: No. We would need to plumb it through, so would just add more complexity, IMO.

Comment (arnavgarg1) [reply tests/ludwig/augmentation/test_augmentation_pipeline.py:149]: Yup, I was thinking plumbing it through, but this is also ok for now

**Pull request merged**